functions/services/pdf_generator.py

```
@@ -0,0 +1,647 @@
+            version="v4",
+            expiration=3600,
+            method="GET",
+        )
+
+        logger.info(
+            "storage_upload",
+            path=storage_path,
+            size=len(pdf_bytes),
+        )
+
+        return download_url, f"gs://{bucket.name}/{storage_path}"
```

Ruff (0.14.8)

Consider moving this statement to an `else` block

(TRY300)